File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        s,e=0,len(nums)-1
        max=0
        while(s<e):
            if(nums[s]+nums[e]>max):
                max=nums[s]+nums[e]
            s+=1
            e-=1
        return max

File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.java
class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int s=0, e=nums.length-1;
        int max=0;
        while(s<e){
            if(max<nums[s]+nums[e])max=nums[s]+nums[e];
            s++;
            e--;
        }
        return max;
    }
}